packages/cozy-pouch-link/src/startReplication.js

```diff
@@ -208,3 +228,55 @@ export const replicateAllDocs = async ({ db, baseUrl, doctype, storage }) => {
   }
   return docs
 }
+
+/**
+ * Replicates all documents from a shared drive to a local PouchDB instance.
+ * This function fetches documents in batches from a Cozy Cloud shared drive
+ * and replicates them to the local database, maintaining replication state
+ * to allow for incremental updates.
+ *
+ * @param {Object} params - The parameters object
+ * @param {string} params.driveId - The unique identifier of the shared drive to replicate from
+ * @param {Object} params.pouch - The local PouchDB instance to replicate documents to
+ * @param {Object} params.storage - Storage interface for persisting replication state
+ * @param {string} params.doctype - The document type being replicated (e.g., 'io.cozy.files')
+ * @param {CozyClient} params.client - CozyClient instance for fetching remote documents
+ * @returns {Promise<Array>} A promise that resolves to an array of all replicated documents
+ * @throws {Error} Throws an error if driveId is not provided
+ */
```

Comment:
Why is this necessary to make this function? When we discussed about it, I  understood that we can call the pouch/couch replication for the shared drive, is it not the case @nono ?

Reply:
Because we don't the paths of the files inside CouchDB, so we can't just make a request to CouchDB to get the relevant data, we need to iterate on each file, get its parent, compute its path, see if it's the shared drive, etc.

Reply:
But isn't it what we already do, to get the change feed for each shared file?

Reply:
No, to get a shared drive's changes feed we filter the main changes feed but don't have to iterate over every file.